### DISCUSSION TITLE
Welcome to openrowingmonitor Discussions!

### DISCUSSION THREAD

**Discussion by JaapvanEkris** [JaapvanEkris/openrowingmonitor#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by gvcormac** [comment]
Hi JaapvanEkris.  We met on the C2 forum.  Is there a howto on hardware for speed sensor?

thanks,
Gordon

**Comment by JaapvanEkris** [comment]
Hi Gordon,

I recognised the alias :) Sorry for the late reply. Most rowing machines use a Hall Effect Sensor combined with two to four magnets on the flywheel (they must use at least two to keep the flywheel balance). Dave Vernooy used a CD with an optical sensor: https://dvernooy.github.io/projects/ergware/

On both the NordicTrack and Concept2 I used the existing sensor.

Jaap

**Comment by ITully** [comment]
Hi,

We have success, first checked that the data was clean, and I am using a KY-024 sensor on DO.  It wasn't clean so moved the sensor back from the magnet and modified the line in GpioTimerService.js " const sensor = new Gpio(config.gpioPin, 'in', 'falling', {debounceTimeout:5})".  I am using an old infiniti Power Row machine.

![power row](https://user-images.githubusercontent.com/38179339/221382518-76cf5606-f71d-4bc1-8f3a-835f2c03d237.png)

Once data was clean, then it was just a matter of getting the configuration right.

A few things to note,

- watch the comment fields and make sure there is a comma after the settings you add except for the last line.
- Kill npm then restart the program by  going to  "root@rowingmonitor:/opt/openrowingmonitor/bin#"
- then invoking  ./openrowingmonitor.sh
- make a change to config.js then ctrl C then run the command line again as above
- You will see the debug messages get written

I am using the app EXR and the BLE is working fine.

**Comment by sean808080** [comment]
hi @JaapvanEkris 
I just upgraded to the v1beta_updates branch at abasz's recommendation since I was having some heart rate reading issues.  Very nice to see so much new dev on your fork here.  Looking forward to learning about all the new functions here.  Cheers
